doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,299 @@
+        &K_i = DRKey\,(AS\,A_i \rightarrow AS_0:Endhost)\,or\,(AS\,A_i \rightarrow AS_0)\\\\
+        &HVF_i = MAC_{K_i}(Identifier, ingress_i, egress_i, encryptedPolicyID_i,\\& srcAddrLen, srcHostAddr)[0:3] \oplus 0x3FFFFF\\\\
+        &HVFVerified_i = MAC_{K_i}(Identifier, ingress_i, egress_i, encryptedPolicyID_i,\\& srcAddrLen, srcHostAddr)[3:6] \oplus 0x3FFFFF\\\\
+    \end{align*}
+
+Data plane
+----------
+
+Processing at the router
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Whenever a FABRID enabled router receives a SCION packet, it has to figure out whether it should be processed as FABRID or not.
+In both cases, all the logic of a normal SCION packet will be applied too.
+The router determines whether the SCION packet is a FABRID packet as follow:
+
+.. image:: fig/FABRID/FABRIDActivation.png
+    :scale: 70%
+
+If the SCION packet uses FABRID, the router is going to verify the correctness of the current FABRID Hop-validation-field using
+either the AS-to-AS or AS-to-Host DRKey and verifies whether the encrypted policy index matches a valid FABRID policy.
+If this is the case, the router will update the FABRID Hop-validation-field accordingly and route the packet over
+an intra-AS path matching the provided FABRID policy.
+The corresponding intra-AS paths are provided to the border routers by the local control service.
+
+Processing at the endhost
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To be able to send a FABRID packet, the endhost has to choose a path that supports its path constraints.
+Then it can request the necessary DRKeys from its local control service.
+With this the endhost is able to create FABRID packets and then send them to the border router for furhter forwarding.
+The receiving endhost can then recompute the path validator to verify that the packet was forwarded over this path.
+
+Control plane
+---------------
+
+Control service
+^^^^^^^^^^^^^^^^^
+
+The control plane for FABRID is responsible for parsing FABRID policies into corresponding data structures, and making
+these datastructures available to routers in the same AS, clients in the AS, as well as clients from a remote AS.
+Information for FABRID is also included in the PCBs emitted by each FABRID enabled AS.
+
+The control service introduces a FABRID service with the following endpoints:
+
+- GetMPLSMapIfNecessary (intra-AS)
+- GetRemotePolicyDescription (intra-AS)
+- GetSupportedIndicesMap (inter-AS, intra-AS)
+- GetIndexIdentifierMap (inter-AS, intra-AS)
+- GetLocalPolicyDescription (inter-AS, intra-AS)
+
+Important data structures
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The FABRID service uses the following important data structures:
+
+- SupportedIndicesMap
+    Maps a connection pair consisting of two ConnectionPoints (Type: string, IP: string, Prefix: uint32, InterfaceId: uint16) to a list of policy indices. This map shows for each connection pair which policy indices are supported, which can be one or multiple policies.
```

Comment:
Would be good to mention that a connection point is either
- an interface
- an IP range
- wildcard 
And that policies can be supported on: interface-interface connections, as well as interface-ip range (on the last hop).

Reply:
I added this additional information